src/router/index.js

```diff
@@ -1,24 +1,35 @@
-import { parse, stringify } from "query-string";
+const storeAndQuery = rawQuery => {
+  const params = decodeURIComponent(rawQuery)
+    .replace(/^\?/, "")
+    .split("&");
+
+  const query = params.filter(pair => !pair.startsWith("storeData=")).join("&");
+  const storeDatas = params.filter(pair => pair.startsWith("storeData="));
+
+  return {
+    query,
+    storeData: (storeDatas[0] && storeDatas[0].replace(/^storeData=/, "")) || {}
```

Comment:
When `storeData` is not found, it should return a JSON string of an empty object (`'{}'`), not an empty object, else the `extractStoreFromLocation()` will blow up. You're not seeing this error because you're catching the parsing error in the extractor function.

Additionally, I think the `storeData` returned value might have to be decoded before returned here, else `JSON.parse()` might attempt to parse a URI encoded value and failing.

Reply:
so i'm going to change this line so it can simply return undefined if no storeData param is found. The storeData  does get decoded in the `extractStoreFromLocation` function.